### PR TITLE
Update Magento UI variables and fix issue #133

### PR DESCRIPTION
**Pull request by RetroModernDev**
I fixed issues with variables in Magento UI _forms.scss also i added missing dependency gulp-sass-error to package.json. Without this dependency command `gulp tests --ci` threw errors.

### CODE REVIEW

File: styles/vendor/magento-ui/_forms.scss [hunk 1]
```diff
@@ -36,37 +36,70 @@ $form-element-input-placeholder__font-style: $form-element-input__font-style !de
...
+$note-icon__font-margin                    : inherit !default;
+$note-icon__font-vertical-align            : $icon-font__vertical-align !default;
+$note-icon__font-position                  : $icon-font__position !default;
+$note-icon__font-text-hide                 : $icon-font__text-hide !default;
+
 @mixin lib-form-element-input($_type) {
-    $_background                : $color-white;
-    $_border                    : 1px solid $form-element-input__border-color;
-    $_border-radius             : 1px;
+    $_background                : $form-element-input__background;
```
Comment (Igloczek) [inline styles/vendor/magento-ui/_forms.scss]: We can't take `$form-element-input__*` as default value for every type of element.
Let's check how the LESS code looks like - https://github.com/magento/magento2/blob/develop/lib/web/css/source/lib/_forms.less

They are using variable `$_type` to create a new variable name for every type. 
In SASS using same structure is prohibited, so if we had to create a set of variables for every type of input manually.

File: .test/package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "gulp-plumber": "~1.1.0",
     "gulp-postcss": "~6.3.0",
     "gulp-sass": "~3.1.0",
+    "gulp-sass-error": "^1.0.4",
```
Comment (Igloczek) [inline .test/package.json:11]: Thanks for that ❤️

**Comment by RetroModernDev** [comment]
Ok, i moved all forms variables to variables/_forms.scss file and i fixed issue with $type variable. 

Additionally I updated variables for other mixins (lib-form-fieldset, lib-form-field, lib-form-hasrequired)

**Comment by Igloczek** [comment]
Good job, thank you!

**Pull request merged**